src/app/dao/tests/PartDAO.spec.ts

```diff
@@ -60,43 +74,249 @@ describe('PartDAO', () => {
             expect(result).toBeFalse();
 
         });
         afterEach(async() => {
-            await signOut(TestBed.inject(Auth));
+            await signOut();
         });
     });
+    it('should allow a verified user to create a part', async() => {
+        // Given a verified user
+        await createConnectedGoogleUser('foo@bar.com', 'creator');
+        // When creating a part
+        const result: Promise<string> = partDAO.create(PartMocks.INITIAL);
+        // Then it should succeed
+        await expectAsync(result).toBeResolved();
+
+    });
+    it('should forbid a non-verified user to create a part', async() => {
+        // Given a non-verified user
+        const token: string = '{"sub": "bar@bar.com", "email": "bar@bar.com", "email_verified": false}';
+        const credential: FireAuth.UserCredential =
+            await FireAuth.signInWithCredential(TestBed.inject(FireAuth.Auth),
+                                                FireAuth.GoogleAuthProvider.credential(token));
+        await userDAO.set(credential.user.uid, { verified: false, username: 'user' });
+
+        // When creating a part
+        const result: Promise<string> = partDAO.create(PartMocks.INITIAL);
+        // Then it should fail
+        await expectFirebasePermissionDenied(result);
+    });
+    it('should forbid player to change playerZero/playerOne/beginning once a part has started', async() => {
+        // Given a part that has started (i.e., turn >= 0), and a player (here creator)
+        const creatorUser: FireAuth.User = await createConnectedGoogleUser('foo@bar.com', 'creator');
+        const creator: MinimalUser = { id: creatorUser.uid, name: 'creator' };
+        await partDAO.create({
+            ...PartMocks.INITIAL,
+            turn: 1,
+            playerZero: creator,
+            playerOne: UserMocks.OPPONENT_MINIMAL_USER,
+        });
+
+        const updates: Partial<Part>[] = [
+            { playerZero: UserMocks.OPPONENT_MINIMAL_USER },
+            { playerOne: creator },
+            { beginning: serverTimestamp() },
+        ];
+        for (const update of updates) {
+            // When trying to change the field
+            const result: Promise<void> = partDAO.update('partId', update);
+            // Then it should fail
+            await expectFirebasePermissionDenied(result);
+        }
+    });
+    it('should forbid non-player to change writable fields', async() => {
+        // Given a part, and a user who is not playing in this game
+        await createConnectedGoogleUser('foo@bar.com', 'creator');
+        await partDAO.create(PartMocks.INITIAL);
+        await signOut();
+        await createConnectedGoogleUser('bar@bar.com', 'non-player');
+
+        const updates: Partial<Part>[] = [
+            { lastUpdate: { index: 1, player: 0 } },
+            { turn: 42 },
+            { result: 3 },
+            { listMoves: [{ a: 1 }] },
+            { lastUpdateTime: serverTimestamp() },
```

Comment:
un moyen de check que celle là peut pas être poussée autrement que par un FieldValue (genre pousser { seconds: 1234567890123456789012356789: nanoseconds: 0 } :p

Reply:
à checker, TODO (mais pas sûr)

Reply:
checké, il semble que t'envoie la requête à Firebase avec FieldValue, et lui la reçoit avec un timestamp, donc non pas moyen de s'assurer de mieux ici !

Reply:
Donc, c'est 100% faisable de dire "mon lastUpdateTime est dans 1000 secondes, donc retire -1000 à mon temps de jeu :p ?

Reply:
Oui :see_no_evil: À améliorer lors de l'ajout du backend !

Reply:
Bah à choisir qu'entre ces deux là, pour sûr, le backend. 
On peut pas faire un SecurityRules.and.return(true) :D ?

Reply:
j'aimerais bien mais non :D Ajouté au ticket backend!

Reply:
Ceci dit on peut faire un check entre les temps utilisés, temps ajoutés (qui seraient des field à ajouter, lol) et le beginning time et checker que ça fait sens ? 
Mais p-e d'abord le backend et d'abord le refactor avec moveAndRequest comme subcollection ?

Reply:
oui ! Ajouté au ticket backend, ça sera moultement plus simple là ! (suffirait de dire « ajoute 30 secondes » et le backend s'occuperais du calcul)